avm-transpiler/scripts/compile_then_transpile.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+set -eu
+
+NARGO=${NARGO:-nargo}
+TRANSPILER=${TRANSPILER:-avm-transpiler}
+
+if [ "${1:-}" != "compile" ]; then
+	echo "Usage: $0 compile"
+	exit 1
+fi
+shift # remove the compile arg so we can inject --show-artifact-paths
+
+# Create a temporary file to capture and parse nargo's stdout while still printing it to the console.
+# To avoid a situation where the script fails and the temporary file isn't removed,
+# create file descriptors for writing/reading the temporary file and then remove the file.
+tmpfile=$(mktemp)
+exec 3>"$tmpfile"
+exec 4<"$tmpfile"
+rm "$tmpfile"
+
+# Forward all arguments to nargo, tee output to the tmp file
+echo "Running nargo ($NARGO compile --show-artifact-paths) with args: $@"
+$NARGO compile --show-artifact-paths "$@" | tee /dev/fd/3
```

Comment:
This feels ugly. Wasn't sure a better/cleaner way to capture nargo's output while leaving it alone as stdout.